libraries/AP_HAL/I2CDevice.h

```diff
@@ -17,6 +17,7 @@
 #pragma once
 
 #include <inttypes.h>
+#include <vector>
```

Comment:
should we avoid to use std::vector on STM32 ?

Reply:
It's not implemented except for Linux boards. I don't see a problem with using it for initialization code. If there's one container type that is sane in stdc++ is vector.